src/game/server/entities/character.cpp

```diff
@@ -1226,31 +1223,68 @@ void CCharacter::FireWeapon()
 			}
 			else
 			{
-				CProjectile *pProj = new CProjectile(GameWorld(), WEAPON_GRENADE,
-					m_pPlayer->GetCID(),
-					ProjStartPos,
-					Direction,
-					(int)(Server()->TickSpeed()*GameServer()->Tuning()->m_GrenadeLifetime),
-					1, true, 0, SOUND_GRENADE_EXPLODE, WEAPON_GRENADE);
-
-	/* INFECTION MODIFICATION START ***************************************/
-				if(GetClass() == PLAYERCLASS_NINJA)
+				if(m_HasStunGrenade) 
+				{
+					int ShotSpread = 2;
+					
+					CMsgPacker Msg(NETMSGTYPE_SV_EXTRAPROJECTILE);
+					Msg.AddInt(ShotSpread*2+1);
+					
+					for(int i = -ShotSpread; i <= ShotSpread; ++i)
+					{
+						float a = GetAngle(Direction) + random_float()/3.0f;
+						
+						CScatterGrenade *pProj = new CScatterGrenade(GameWorld(), m_pPlayer->GetCID(), m_Pos, vec2(cosf(a), sinf(a)));
+						
+						if (m_HasStunGrenade)
+						{
+							//Make them flash grenades
+							pProj->FlashGrenade();
+						}
+						
+						
+						// pack the Projectile and send it to the client Directly
+						CNetObj_Projectile p;
+						pProj->FillInfo(&p);
+						
+						for(unsigned i = 0; i < sizeof(CNetObj_Projectile)/sizeof(int); i++)
+							Msg.AddInt(((int *)&p)[i]);
+						Server()->SendMsg(&Msg, MSGFLAG_VITAL, m_pPlayer->GetCID());
+					}
+					
+					GameServer()->CreateSound(m_Pos, SOUND_GRENADE_FIRE);
+					
+					m_HasStunGrenade=false;
+					GetPlayer()->ResetNumberKills();
```

Comment:
why are we sure that GetPlayer() doesn't return null here?

Reply:
As I see it, GetPlayer() is a method from a characters object, if there would not be a character there would be no player to call.
But that also means if there is no character, this object can't possibly call GetPlayer() resulting in a Null.